quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,395 @@
+        clientRequestProperties.setClientRequestId(String.format("%s;%s", scope, UUID.randomUUID()));
+
+        // Tip: While not common, you can alter the request default command timeout using the below command, e.g. to set the timeout to 10 minutes, use "10m"
+        if (StringUtils.isNotBlank(timeout)) {
+            clientRequestProperties.setOption(OPTION_SERVER_TIMEOUT, timeout);
+        }
+        return clientRequestProperties;
+    }
+
+    private static void ingestDataFromFile(IngestClient ingestClient, String databaseName, String tableName, String uri, IngestionProperties.DataFormat dataFormat, String mappingName) {
+        IngestionProperties ingestionProperties = createIngestionProperties(databaseName, tableName, dataFormat, mappingName);
+
+        // Tip 1: For optimal ingestion batching and performance, specify the uncompressed data size in the file descriptor.
+        //  Otherwise, the service will determine the file size, but it will require an additional s2s call, and may not be accurate for compressed files.
+        // Tip 2: To correlate between ingestion operations in your applications and Kusto, set the source id and log it somewhere.
+
+        File fileToIngest = new File(".\\" + uri);
+        FileSourceInfo fileSourceInfo = new FileSourceInfo(fileToIngest.getPath(), fileToIngest.length(), UUID.randomUUID());
+        try {
+            ingestClient.ingestFromFile(fileSourceInfo, ingestionProperties);
+        } catch (IngestionClientException e) {
+            System.out.printf("Client exception while trying to ingest '%s' into '%s.%s'%n%n", uri, databaseName, tableName);
+            e.printStackTrace();
+        } catch (IngestionServiceException e) {
+            System.out.printf("Service exception while trying to ingest '%s' into '%s.%s'%n%n", uri, databaseName, tableName);
+            e.printStackTrace();
+        }
+
+        // Tip: Kusto's Java SDK can ingest data from files, blobs, java.sql.ResultSet objects, and open streams.
+        //  See the SDK's kusto-samples module and the E2E tests in kusto-ingest for additional references.
+    }
+
+    private static void ingestDataFromBlob(IngestClient ingestClient, String databaseName, String tableName, String blobUrl, IngestionProperties.DataFormat dataFormat, String mappingName) {
+        IngestionProperties ingestionProperties = createIngestionProperties(databaseName, tableName, dataFormat, mappingName);
+
+        // Tip 1: For optimal ingestion batching and performance, specify the uncompressed data size in the file descriptor instead of the default below of 0.
+        //  Otherwise, the service will determine the file size, but it will require an additional s2s call, and may not be accurate for compressed files.
+        // Tip 2: To correlate between ingestion operations in your applications and Kusto, set the source id and log it somewhere
+        BlobSourceInfo blobSourceInfo = new BlobSourceInfo(blobUrl, 0, UUID.randomUUID());

```

Comment:
Follow the above comment, perhaps the size should come from the outside (as 0 or default = 0)

Reply:
My intent was to demonstrate both approaches.
I'll consider ways to address your request in the previous comment.